### PR TITLE
[ADD] Geolocation: Search Results

### PR DESCRIPTION
**Pull request by parkerthegeniuschild**
#### What does this PR do?
Allows user to find products within a specified radius in km

#### How should this be manually tested?
GET /products

#### Any background context you want to provide?
The geolocation uses the same endpoint as the regular search but when queries are added:
`lat, lng, radius`; it returns products only available within the radius and also informs the user
of the distance to each product e.g.

Radius should be an integer measured in `km`.

#### Screenshots
---
image
---
image

### CODE REVIEW

File: src/utils/helpers.utils.js [hunk 1]
```diff
@@ -66,4 +67,51 @@ export default {
...
+        * Math.sin(differenceInLatitude / 2)
+        + Math.cos(convertDegreeToRadian(userLatitude))
+        * Math.cos(convertDegreeToRadian(productLatitude))
+        * Math.sin(differenceInLongitude / 2)
+        * Math.sin(differenceInLongitude / 2);
+
+      const y = 2 * Math.atan2(Math.sqrt(x), Math.sqrt(1 - x));
+
+      // Distance in km between the user's location and the product in the current iteration
+      const distance = earthRadiusInKm * y;
```
Comment (mrayor) [inline src/utils/helpers.utils.js]: How did you figure out this formula fam...

**Comment by mrayor** [comment]
You also have merge conflicts...

File: src/utils/helpers.utils.js (hunk 1 above)
Comment (parkerthegeniuschild) [reply src/utils/helpers.utils.js]: pure maths and geography . . see below

image

**Comment by mrayor** [comment]
Yes I know but mehn 🍾🎊🎉

**Comment by parkerthegeniuschild** [comment]
conflicts fixed !

**Pull request merged**